packages/office-ui-fabric-react/src/components/Nav/Nav.base.tsx

```diff
@@ -105,13 +105,13 @@ export class NavBase extends React.Component<INavProps, INavState> implements IN
 
   private _renderNavLink(link: INavLink, linkIndex: number, nestingLevel: number): JSX.Element {
     const { styles, groups, theme, onRenderLink = this._onRenderLink, linkAs: LinkAs = ActionButton } = this.props;
-
+    const isLinkWithIcon = link.iconProps && (!!link.iconProps.iconName || !!link.iconProps.imageProps);
     const classNames = getClassNames(styles!, {
       theme: theme!,
       isSelected: this._isLinkSelected(link),
       isDisabled: link.disabled,
       isButtonEntry: link.onClick && !link.forceAnchor,
-      leftPadding: _indentationSize * nestingLevel + _baseIndent,
+      leftPadding: _indentationSize * nestingLevel + _baseIndent + (isLinkWithIcon ? 0 : 24),
```

Comment:
Where does the `24` come from? Is it the size of an icon? If so, would be nice to pull that constant from icon. If not, should at least be defined as a constant near the top of this file.

Reply:
This is the width of the empty icon used as left-padding when omitted. I tried to think of how to do this w/o hard-coding, but can only think of either:

1) Measure icon element once rendered which will incur a cost
2) Do some calculation based on font-size and other factors and export

Scenario I can think of where this may need the above dynamic height is where icon is customized and then once omitted alignment is off. But in that case I think custom styling could solve and the perf fix would be beneficial enough to warrant it. Defer to you ultimately as I'm less familiar with scenarios.

We should avoid rendering this component since its expensive for what it is, might be worth this fix or a prop to opt-in.